tree/dataframe/src/RDFColumnRegister.cxx

```diff
@@ -40,8 +40,16 @@ RDefineReader *RDefinesWithReaders::GetReader(unsigned int slot, const std::stri
    if (variationName != "nominal")
       define = &define->GetVariedDefine(variationName);
 
+#if !defined(__clang__) && __GNUC__ >= 7 && __GNUC_MINOR__ >= 3
    const auto insertion = defineReaders.insert({variationName, std::make_unique<RDefineReader>(slot, *define)});
-   return insertion.first->second.get();
+   auto *ret = insertion.first->second.get();
```

Comment:
Why did you change this? Why not just `return ret` for the old case? Once we remove the old GCC case we would be left with this extra return...

Reply:
it seems that I have an ingrained dislike for multiple return statements. even when only one gets to be compiled. i'll fix it, one sec.

Reply:
it should be ok now